### DISCUSSION TITLE
TVDB account plugin

### DISCUSSION THREAD

**Discussion by techiegirl2day** [OpenSageTV/Community#26] [Gemstone]
Has a plugin been made available yet to address the coming paid TVDB user subscriptions?  
I ask because I would like to restore/replace the show metadata we lost from IMDB changes.

**Comment by jusjoken** [comment]
If you use the most recent version of Gemstone with the most recent version of Phoenix Core, then the TVDB provider has already been updated to the latest version that is typically a paid for subscription.  I was able to convince TVDB to provide the SageTV community users free access so just using Gemstone/Phoenix core latest versions will provide this functionality.

Does that make sense?

Check out [this post](https://forums.sagetv.com/forums/showthread.php?t=67091) on the old forums for more info...

**Reply by Sagetvmaster** [reply]
and if you don't use Gemstone/Phoenix?  No go?

**Reply by techiegirl2day** [reply]
Ok thanx. I hadn't checked the Gemstone & Phoenix cores.  I will look & see what I have there.

**Reply by jusjoken** [reply]
Your defaults will use tvdb4 so no need to change anything

**Reply by PhilipCRoberts** [reply]
It would be nice to change the IMDB Search shortcut in the Options menu to TVDB.

**Reply by jusjoken** [reply]
Not sure what you are referring to specifically...the search menu where ?

**Reply by PhilipCRoberts** [reply]
Options 

Watch Now
Record Options
View Recording Detail
IMDB Search <-- This option is broken. Can TVDB live here?

**Reply by techiegirl2day** [reply]
Ditto I second that request.  It goes with the CMT plugin.

**Reply by jusjoken** [reply]
I see.  IT would not be a small amount of work but as IMDB access is now only available as a paid service and that rewrite would also be large, using TheTVDB makes sense as the community has the rights to access the data free through the API.

I can look into it as time permits but it may not be able to be an inline replacement for IMDB but may need to be another UI plugin.

---
@jusjoken
If you wish to see what I am up to and support my efforts visit my [Patreon page](https://www.patreon.com/user?u=58035567)

**Reply by Sagetvmaster** [reply]
> I see. IT would not be a small amount of work but as IMDB access is now only available as a paid service and that rewrite would also be large, using TheTVDB makes sense as the community has the rights to access the data free through the API.
> 
> I can look into it as time permits but it may not be able to be an inline replacement for IMDB but may need to be another UI plugin.
> 
> @jusjoken If you wish to see what I am up to and support my efforts visit my [Patreon page](https://www.patreon.com/user?u=58035567)

Now that IMDB became a paid, content provider, I'd expect them to be changing up code quite often, so yeah, probably not worth the time and expense to chase it if TVDB has done us a solid with the FREE access.  Yeah?

**Comment by techiegirl2day** [comment]
Okay, I went into the Phoenix core & see 3 options for metadata providers?  Is this what you were referring to? 
 Is tvdb 4 the one I want enter subscription details (when they make the service available)?

**Reply by jusjoken** [reply]
Yes.  Its tvdb 4.  You do not need to enter any subscription info.. It will just use tvdb4 by default now if the provider is tvdb or tvdb4.

**Reply by Sagetvmaster** [reply]
Hi.  I'm looking at the BMT config page, and select 'Phoenix/The TVDB4 Configuration', it's just a blank page.  Is this normal?  If not, what am I missing, aka what should populate this page.
I have Phoenix Core and BMT tools installed and all is up to date.  Is there a quality link on the old site that would help in troubleshooting this?
That BMT page you linked earlier - I'm not finding that url yet, the page with the TVDB4 logo in the corner

**Reply by jusjoken** [reply]
Yes that's normal...that page was intended for subscription info BEFORE I was able to request we get free access...so blank is normal.

The image in the link I posted above is from the Browse function in BMT when a TV item is selected and the provider is set to tvdb.

---
@jusjoken
If you wish to see what I am up to and support my efforts visit my [Patreon page](https://www.patreon.com/user?u=58035567)

**Reply by Sagetvmaster** [reply]
I’m thinking my installation isn’t right then?  I found this line in the link:
“ U should see a link at the top that says TheTVDB.”
I don’t see and link or reference to TheTVDB on the ‘browse’ page of BMT.
PS: I’m not sure I EVER had a subscription or PIN for tvdb. But there is a “tvdb4/token=“ in my sage.properties file…

**Reply by jusjoken** [reply]
Can you post a screen shot to help me understand

**Reply by Sagetvmaster** [reply]
> Can you post a screen shot to help me understand
This is what my ‘browse’ window looks like if the ‘search’ box is left empty
![IMG_2507](https://github.com/OpenSageTV/Community/assets/133995114/39134add-c8ef-4e83-854d-1205bacbfe34)

**Reply by jusjoken** [reply]
It should look more like this with a list of views....

![image](https://github.com/OpenSageTV/Community/assets/6617820/8f7a376e-2a7d-4da9-990b-56866ccb2615)

**Reply by Sagetvmaster** [reply]
Hi.  Thanks for clarifying that.  Something’s obviously not right with mine.  Suggestions?  Am I missing a jar file, etc.?  Should I just uninstall / install Phoenix Core & BMT again?  Both claim to be the updated versions.
On a similar note, my Android phone installation of ‘Phoenix Remote’ is also showing ‘no media’ in the ‘managed views’ page.  Possibly these events are related?
There is a tvdb4 token listed in my Sage.properties file, but no reference to an earlier ‘PIN’
![IMG_2510](https://github.com/OpenSageTV/Community/assets/133995114/11e0ade1-4d22-4e7a-946a-f53e289935e7)
![IMG_2509](https://github.com/OpenSageTV/Community/assets/133995114/6e2ad61f-8ab5-412d-a942-2029e771e0c3)

**Reply by jusjoken** [reply]
I would say yes they are related.  Can you post your phoenix.log file from the logs folder after going to the BMT browse and I will review.

**Reply by Sagetvmaster** [reply]
Thanks, here you go:
[phoenix.log](https://github.com/OpenSageTV/Community/files/11557626/phoenix.log)

**Reply by jusjoken** [reply]
The log indicated the plugin dependency dom4j is not found.  Check if that plugin (library plugin) is installed and must be version 1.6.2 dated March 8th 2023

---
@jusjoken
If you wish to see what I am up to and support my efforts visit my [Patreon page](https://www.patreon.com/user?u=58035567)

**Reply by Sagetvmaster** [reply]
It was modified on 3/8/23 is 1.6.2 and appears to be installed, but there are TWO folders in the PluginInstallers\dom4j dir: 1.6.1 and 1.6.2, and both are empty folders

**Reply by jusjoken** [reply]
Perhaps provide a list of jar files in the JARS folder as that plugin is failing according to the log.

**Reply by Sagetvmaster** [reply]
> Perhaps provide a list of jar files in the JARS folder as that plugin is failing according to the log.

Is the jar supposed to be named 1.6.2 as well?  'Cause I don't see it in there.  1.6.1 was there, but it had been renamed with a numerical suffix, I imagine since 1.6.2 was supposed to be there at some point?

UPDATE: I renamed 1.6.1 so the suffix was .jar as opposed to .jar.300 or whatever it was.  All the Browse/View items have returned on the left side of the frame.   Also, all of of my 'Views' in the Phoenix Remote app have populated again.  Fix was that simple it would seem.  Just let me know if the 1.6.1 jar should be the later version, 1.6.2.  Thanks for the assist :)

**Reply by jusjoken** [reply]
The 1.6.1 and 1.6.2 jars are identical.  The version was only changed to force it to be downloaded from am available location.

**Reply by Sagetvmaster** [reply]
> The 1.6.1 and 1.6.2 jars are identical. The version was only changed to force it to be downloaded from am available location.

Good news then.  Thanks.  Time to check the other jars in the folder which have been numerically suffixed…

**Reply by techiegirl2day** [reply]
Glad I asked. Will check my files too.
